src/test/java/uk/gov/ons/ctp/response/collection/exercise/endpoint/CollectionExerciseEndpointIT.java

```diff
@@ -1,162 +1,344 @@
+
+    assertEquals(201, (int) result.getLeft());
+    CollectionExerciseDTO newCollex = this.client.getCollectionExercise(result.getRight());
+
+    log.info("Collection exercise to link: {}", newCollex);
+
+    UUID sampleSummaryId = UUID.randomUUID();
+
+    final int status = this.client.linkSampleSummary(newCollex.getId(), sampleSummaryId);
+    assertEquals(200, status);
+
+    SimpleMessageSender sender = getMessageSender();
+
+    SampleSummaryDTO sampleSummary = new SampleSummaryDTO();
+    sampleSummary.setId(sampleSummaryId);
+
+    SimpleMessageListener listener = getMessageListener();
+    BlockingQueue<String> queue = listener.listen(SimpleMessageBase.ExchangeType.Direct,
+                                                  "collection-outbound-exchange", "SampleLink.Activated.binding");
+
+    // This will cause an exception to be thrown as there is no collection instrument service but this is
+    // harmless to our purpose
+    sender.sendMessage("sample-outbound-exchange", "Sample.SampleUploadFinished.binding",
+                       this.mapper.writeValueAsString(sampleSummary));
+
+    queue.take();
+
+    List<SampleLinkDTO> links = this.client.getSampleLinks(newCollex.getId());
+
+    assertEquals(1, links.size());
+
+    SampleLinkDTO link = links.get(0);
+
+    assertEquals(sampleSummaryId, UUID.fromString(link.getSampleSummaryId()));
+    assertEquals("ACTIVE", link.getState());
+  }
+
+  @Test
+  public void ensureSampleUnitIdIsPropagatedHere() throws Exception {
+    createCollectionInstrumentStub();
+    createSurveyServiceSocialStub();
+
+    SampleUnit sampleUnit = new SampleUnit();
+    UUID id = UUID.randomUUID();
+    SimpleMessageSender sender = getMessageSender();
+
+    CollectionExerciseDTO collex = createCollectionExercise(TEST_SURVEY_ID, getRandomRef(), "Test description");
+
+    sampleUnit.setId(id.toString());
+    sampleUnit.setSampleUnitRef("LMS0001");
+    sampleUnit.setCollectionExerciseId(collex.getId()
+                                             .toString());
+    sampleUnit.setFormType("");
+    sampleUnit.setSampleUnitType("H");
+    //sampleUnit.setSampleAttributes(new SampleUnit.SampleAttributes(new ArrayList<>()));
+
+    setSampleSize(collex, 1);
+    setState(collex, CollectionExerciseDTO.CollectionExerciseState.EXECUTION_STARTED);
+
+    SimpleMessageListener listener = getMessageListener();
+    BlockingQueue<String> queue = listener.listen(SimpleMessageBase.ExchangeType.Direct,
+                                                  "collection-outbound-exchange", "Case.CaseDelivery.binding");
+
+    String xml = sampleUnitToXmlString(sampleUnit);
+
+    log.debug("xml = " + xml);
+
+    sender.sendMessage("sample-outbound-exchange", "Sample.SampleDelivery.binding",
+                       xml);
+
+    // The service stubs will exit once the call below times out preventing further debugging of this test in other
```

Comment:
Not sure if this comment is necessary if we're using queue.take() in the test normally?